src/services/charmcraft/charmcraft.ts

```diff
@@ -25,18 +25,18 @@ class Charmcraft {
 
   async uploadResources() {
     let resourceInfo = 'resources:\n';
-    if (!this.uploadImage) {
-      const msg =
-        `No resources where uploaded as part of this build.\n` +
-        `If you wish to upload the OCI image, set 'upload-image' to 'true'`;
-      core.warning(msg);
-      return { flags: [''], resourceInfo: '' };
-    }
-
     const { name, images } = this.metadata();
+
     const flags = await Promise.all(
       images.map(async ([resource_name, resource_image]) => {
-        await this.uploadResource(resource_image, name, resource_name);
+        if (!this.uploadImage) {
+          const msg =
+            `No resources where uploaded as part of this build.\n` +
+            `If you wish to upload the OCI image, set 'upload-image' to 'true'`;
+          core.warning(msg);
+        } else {
+          await this.uploadResource(resource_image, name, resource_name);
+        }
```

Comment:
```suggestion
    if (!this.uploadImage) {
      const msg =
        `No resources where uploaded as part of this build.\n` +
        `If you wish to upload the OCI image, set 'upload-image' to 'true'`;
      core.warning(msg);
    }
    const flags = await Promise.all(
      images.map(async ([resource_name, resource_image]) => {
        if (this.uploadImage) {
          await this.uploadResource(resource_image, name, resource_name);
        }
```

Should probably be this way to avoid log spam?

Reply:
Fixed